### PR TITLE
Simplify gRPC service definition for the worker

### LINKED ISSUE #6074: Clean up gRPC server abstractions in the core worker
We should clean up gRPC in the core worker as follows:
1. There should be a common gRPC core worker handler. The handler methods can either post events onto the main event loop, or call other core worker methods directly.
2. No other interfaces.

This way, a worker->worker and raylet->worker RPCs can be added with fewer changes: (1) defining the proto message, (2) writing the handler method, and (3) the client handling code.

We should also probably add a worker->raylet RPC client. I believe the only thing preventing this currently is plumbing the node manager port into the core worker.

cc @edoakes @stephanie-wang

### PR DESCRIPTION
**Pull request by edoakes**
## Why are these changes needed?

It's currently complicated to add a new endpoint to the worker's gRPC server and also opaque where the endpoint handlers are run. This change makes it clear, as the `CoreWorker` handles all endpoints and posts them to the appropriate `io_service` as needed and also adds a macro so that all C++ changes happen in one place.

## Related issue number

Closes #6074

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.

### CODE REVIEW

File: src/ray/core_worker/core_worker.h
Comment (ericl) [inline src/ray/core_worker/core_worker.h]: How about 9999 for this one?

File: src/ray/core_worker/core_worker.h
Comment (ericl) [inline src/ray/core_worker/core_worker.h]: common rpc service for all core worker modules

File: src/ray/protobuf/worker.proto
Comment (ericl) [inline src/ray/protobuf/worker.proto]: Can you rename this file to core_worker.proto?

**Review by ericl — APPROVED** [review]
LGTM if tests pass

File: src/ray/core_worker/core_worker.h
Comment (pcmoritz) [inline src/ray/core_worker/core_worker.h]: We should avoid using `/* */` comments (it prevents you from being able to comment things out with them as they can't be nested)

File: src/ray/protobuf/worker.proto
Comment (pcmoritz) [inline src/ray/protobuf/worker.proto]: typo

File: src/ray/rpc/worker/worker_client.h
Comment (stephanie-wang) [inline src/ray/rpc/worker/worker_client.h]: Can we put these methods in a .cc?

File: src/ray/core_worker/core_worker.h
Comment (stephanie-wang) [inline src/ray/core_worker/core_worker.h]: Can you be more specific about the naming convention that you need to use for the corresponding code in the protobuf and CoreWorker?

**Review by stephanie-wang — COMMENTED** [review]
Thanks, this is great!

File: src/ray/rpc/worker/worker_client.h
Comment (edoakes) [reply src/ray/rpc/worker/worker_client.h]: Good idea but I tried doing this and got a bunch of thread annotation errors (which I think means the existing thread annotations aren't actually working), so I will leave this to @ericl.

File: src/ray/core_worker/core_worker.h
Comment (edoakes) [reply src/ray/core_worker/core_worker.h]: Done. Added an example for clarity

**Comment by ericl** [comment]
Merging to unblock direct task calls -- feel free to iterate further in parallel.

**Pull request merged**
